Marlin/Marlin_main.cpp

```diff
@@ -1712,8 +1714,12 @@ static void clean_up_after_endstop_or_probe_move() {
     if ((Z_HOME_DIR) < 0 && zprobe_zoffset < 0)
       z_dest -= zprobe_zoffset;
 
-    if (z_dest > current_position[Z_AXIS])
+    if (z_dest > current_position[Z_AXIS]) {
+      float old_feedrate = feedrate;
+      feedrate = homing_feedrate[Z_AXIS];
       do_blocking_move_to_z(z_dest);
+      feedrate = old_feedrate;
+    }
```

Comment:
Do you want to do this for every call to `do_blocking_move_to_z()`? It should bete be __in__  `do_blocking_move_to_z()`.  In fakt it's allreay in - in `do_blocking_move_to()`.